core/Version.php

```diff
@@ -56,4 +56,28 @@ public function isPreviewVersion($version): bool
 
         return false;
     }
+
+    public function nextPreviewVersion($version): string
+    {
+        if (!$this->isVersionNumber($version)) {
+            return '';
+        }
+
+        $dt = date('YmdHis');
+
+        if ($this->isPreviewVersion($version)) {
+            // already a preview, update dt and check it's newer
+            $newVersion = substr($version, 0, -14) . $dt;
+            if (version_compare($version, $newVersion, '<')) {
+                return $newVersion;
+            }
+            return '';
+        } elseif ($this->isStableVersion($version)) {
+            // no suffix yet
+            return $version . '-alpha.' . $dt;
```

Comment:
This seems incorrect. If we e.g. have released Matomo `5.3.0`, the next preview release should not be `5.3.0-alpha.XXXXXX`. We would need to increase the patch (or minor) release number in that case.

Reply:
Hmmmm, you might be onto something here. I checked the docs on the preview release versioning but I didn't think that for `x.y.0` we would need to bump the patch version.
If that's the case (can you confirm @sgiehl and @mneudert) then I can adjust the logic here to check for `x.y.0` versions specifically.

Reply:
It's not only around `x.y.0`. If we have released e.g. 5.3.1, the next would need to be 5.3.2-alpha.

Reply:
Fair point, will adjust.

Reply:
@sgiehl this has been adjusted, please re-review. Thanks!